python/doc/examples/data_analysis/statistical_hypothesis_testing/plot_kolmogorov_statistics.py

```diff
@@ -2,53 +2,70 @@
 The Kolmogorov-Smirnov statistics
 =================================
 """
-# %% 
+# %%
 
 # %%
-# In this example, we illustrate how the Kolmogorov-Smirnov statistics is computed. 
+# In this example, we illustrate how the Kolmogorov-Smirnov statistics is computed.
 #
-# * We generate a sample from a gaussian distribution. 
+# * We generate a sample from a gaussian distribution.
 # * We create a Uniform distribution which parameters are estimated from the sample.
 # * The Kolmogorov-Smirnov statistics is computed and plot on the empirical cumulated distribution function.
 
 # %%
 import openturns as ot
 import openturns.viewer as viewer
 from matplotlib import pylab as plt
+
 ot.Log.Show(ot.Log.NONE)
 
 # %%
 # The computeKSStatisticsIndex function computes the Kolmogorov-Smirnov distance between the sample and the distribution. Furthermore, it returns the index which achieves the maximum distance in the sorted sample. The following function is for teaching purposes only: use `FittingTest` for real applications.
 
 # %%
-def computeKSStatisticsIndex(sample,distribution):
+def computeKSStatisticsIndex(sample, distribution):
     sample = ot.Sample(sample.sort())
+    print("Sorted")
+    print(sample)
     n = sample.getSize()
-    D = 0.
+    D = 0.0
     index = -1
-    D_previous = 0.
+    D_previous = 0.0
     for i in range(n):
         F = distribution.computeCDF(sample[i])
-        D = max(F - float(i)/n,float(i+1)/n - F,D)
-        if (D > D_previous):
+        S1 = abs(F - float(i) / n)
+        S2 = abs(float(i + 1) / n - F)
+        print(
+            "i=%d, x[i]=%.4f, F(x[i])=%.4f, S1=%.4f, S2=%.4f"
+            % (i, sample[i, 0], F, S1, S2)
+        )
+        D = max(S1, S2, D)
+        if D > D_previous:
+            print("D max!")
             index = i
             D_previous = D
-    return D, index
+    observation = sample[index]
+    return D, index, observation
 
 
 # %%
-# The drawKSDistance function plots the empirical distribution function of the sample and the Kolmogorov-Smirnov distance at point x. 
+# The drawKSDistance function plots the empirical distribution function of the sample and the Kolmogorov-Smirnov distance at point x. The empirical CDF is a staircase function and is discontinuous at each observation. The computeEmpiricalCDF() method computes the probability P(X <= x), but this only takes into account for half the extreme values of the CDF. The other half is P(X < x) which is approximated by P(X <= x - delta) where delta is close to zero.
```

Comment:
Could you perhaps write the math expressions in LaTeX?

Reply:
Why not, but this requires edition.

Reply:
If fixed this in the latest commit.